### PR TITLE
Don't install GraalVM for an architecture that can't be built 

### LINKED ISSUE #450: `gw buildNative` runs aarch64 tasks on amd64
On amd64 WSL/Ubuntu, `gw buildNative` now additionally runs `:downloadGraalVmAarch64` (which succeeds) and `:installGraalVmAarch64` (which fails). Neither of these tasks should be run.

### PR DESCRIPTION
**Pull request by bioball**
Fixes an issue where native builds will try to install GraalVM for the wrong architecture.

Closes #450

### CODE REVIEW

File: pkl-cli/pkl-cli.gradle.kts [hunk 1]
```diff
@@ -218,39 +215,48 @@ fun Exec.configureExecutable(
...
  */
 val macExecutableAmd64: TaskProvider<Exec> by tasks.registering(Exec::class) {
-  dependsOn(":installGraalVmAmd64")
-  configureExecutable(
-    buildInfo.os.isMacOsX,
-    buildInfo.graalVmAmd64,
-    layout.buildDirectory.file("executable/pkl-macos-amd64")
-  )
+  if (buildInfo.os.isMacOsX) {
+    dependsOn(":installGraalVmAmd64")
```
Comment (bioball) [inline pkl-cli/pkl-cli.gradle.kts:219]: Curious if there's a better way to describe a conditional task dependency than this?

Comment (translatenix) [reply pkl-cli/pkl-cli.gradle.kts:219]: Why is this dependency conditional?

File: pkl-cli/pkl-cli.gradle.kts [hunk 2]
```diff
@@ -218,39 +215,48 @@ fun Exec.configureExecutable(
...
+    configureExecutable(
+      buildInfo.graalVmAmd64,
+      layout.buildDirectory.file("executable/pkl-macos-amd64")
+    )
+  } else {
+    enabled = false
+  }
```
Comment (holzensp) [inline pkl-cli/pkl-cli.gradle.kts:226]: Shouldn't this just be
```suggestion
  enabled = buildInfo.os.isMacOsX
  dependsOn(":installGraalVmAmd64")
  configureExecutable(
    buildInfo.graalVmAmd64,
    layout.buildDirectory.file("executable/pkl-macos-amd64")
  )
```

? [throughout]

Surely, if `!enabled`, tasks this one `dependsOn` are not executed.

**Review by holzensp — APPROVED** [review]
Non-blocking suggestion for simplification.

File: pkl-cli/pkl-cli.gradle.kts (hunk 1 above)
Comment (bioball) [reply pkl-cli/pkl-cli.gradle.kts:219]: Without the `if`, I am seeing that `installGraalVmAmd64` gets executed on every OS/arch.

Comment (translatenix) [reply pkl-cli/pkl-cli.gradle.kts:219]: Sounds like something else is wrong with the build script. The `if` doesn’t make sense to me.

Comment (bioball) [reply pkl-cli/pkl-cli.gradle.kts:219]: Task `buildNative` depends on `macExecutableAmd64`, `macExecutableAarch64`, etc, etc.

I am seeing that these tasks themselves don't run if `enabled = false`, but its dependent tasks still run, unless I _also_ don't call `dependsOn` at all.

**Pull request merged**